File: src/main/java/com/github/hibi_10000/plugins/plategate/PlateGate.java
package com.github.hibi_10000.plugins.plategate;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

import com.github.hibi_10000.plugins.plategate.util.ArrayJson;
import org.bukkit.plugin.java.JavaPlugin;

import com.github.hibi_10000.plugins.plategate.command.PGBase;
import com.github.hibi_10000.plugins.plategate.event.Event;

public class PlateGate extends JavaPlugin {
	
	public String ver = getDescription().getVersion();
	
	public File gatelistjsonfile = new File(getDataFolder(), "gatelist.json");

	public ArrayJson arrayJson;

	@Override
	public void onEnable() {
		if (!gatelistjsonfile.exists()) {
			saveResource("gatelist.json", false);
		}
		getCommand("pg").setExecutor(new PGBase(this));
		getCommand("pg").setTabCompleter(new PGBase(this));
		getServer().getPluginManager().registerEvents(new Event(this), this);
		List<String> namelist = new ArrayList<>();
		namelist.add("name");
		namelist.add("owner");
		namelist.add("to");
		namelist.add("x");
		namelist.add("y");
		namelist.add("z");
		namelist.add("rotate");
		namelist.add("world");
		namelist.add("beforeblock");
		arrayJson = new ArrayJson(this, namelist);
	}
}

File: src/main/kotlin/com/github/hibi_10000/plugins/plategate/PlateGate.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package com.github.hibi_10000.plugins.plategate

import com.github.hibi_10000.plugins.plategate.command.PGBase
import com.github.hibi_10000.plugins.plategate.database.DBUtil
import com.github.hibi_10000.plugins.plategate.database.JsonUtil
import com.github.hibi_10000.plugins.plategate.event.Event
import org.bukkit.plugin.java.JavaPlugin
import java.io.File

lateinit var instance: PlateGate
lateinit var dbUtil: DBUtil
val util: Util = Util()

class PlateGate : JavaPlugin() {
    override fun onEnable() {
        instance = this
        val gateDB = File(dataFolder, "gate.json")
        if (!gateDB.exists()) {
            saveResource("gate.json", false)
        }
        dbUtil = JsonUtil(gateDB)
        getCommand("plategate")?.setExecutor(PGBase())
        getCommand("plategate")?.tabCompleter = PGBase()
        server.pluginManager.registerEvents(Event(), this)
    }
}
